cpmpy/transformations/linearize.py

```diff
@@ -572,3 +604,26 @@ def only_positive_coefficients(lst_of_expr):
 
     return newlist
 
+def linearize_objective(expr, supported=frozenset(["sum","wsum"])):
+    """
+    - Decision variable: Var
+    - Linear: sum([Var])                                   (CPMpy class 'Operator', name 'sum')
+              wsum([Const],[Var])                          (CPMpy class 'Operator', name 'wsum')
+              no negated BoolVars
+
+    Only difference with flatten_objective is that no negated boolvars are allowed.
+    """
+    flatexpr, flatcons = flatten_objective(expr, supported=supported)
+    pos_expr, const = only_positive_bv_sub(flatexpr)
+    if isinstance(const, list):
+         return pos_expr, flatcons + const
+    pos_expr = Operator("wsum", [pos_expr.args[0]+[1], pos_expr.args[1]+[const]])
```

Comment:
euh... what?

if your function will always return a wsum, you should be more explicit about that.

then, again you create a new copy while it might not be needed...
if const != 0:
  assert pos_expr.name == "wsum", "linearize_obj: expected wsum"
  pos_expr = Operator("wsum", ...